File: llm/lib/tasks/init-prompts.mjs
// init-prompts — seed .kb/config/prompts/ from templates/prompts/.
import { initPrompts } from '../prompts.mjs';

export async function run({ kbRoot, input }) {
  const result = initPrompts(kbRoot, { force: input?.force === true });
  return {
    ok: true,
    src: result.src,
    dst: result.dst,
    prompts: result.results,
  };
}

File: agent/weft_agent/tasks/init_prompts.py
"""init-prompts — seed .kb/config/prompts/ from templates/prompts/
(port of llm/lib/tasks/init-prompts.mjs).
"""
from ..prompts import init_prompts, resolve_prompt


def run(kb_root, input=None, output_path=None):
    result = init_prompts(kb_root, force=bool((input or {}).get("force") is True))
    resolve_prompt.cache_clear()
    return {
        "ok": True,
        "src": result["src"],
        "dst": result["dst"],
        "prompts": result["results"],
    }
